<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>Check for fishy genes</title><link href='https://fonts.loli.net/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext' rel='stylesheet' type='text/css' /><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color:#ffffff; --text-color:#333333; --select-text-bg-color:#B5D6FC; --select-text-font-color:auto; --monospace:"Lucida Console",Consolas,"Courier",monospace; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; bottom: 0px; top: 0px; left: 0px; right: 0px; font-size: 1rem; line-height: 1.42857; overflow-x: hidden; background: inherit; tab-size: 4; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; word-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 40px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
@media screen and (max-width: 500px) {
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:first-child { margin-top: -20px; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; }
button, input, select, textarea { color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: inherit; line-height: inherit; font-family: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 2; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.7); color: rgb(85, 85, 85); border-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px !important; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right: 0px; background-color: inherit; }
.CodeMirror-linenumber { user-select: none; }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background: inherit; position: relative !important; }
.md-diagram-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; background: 0px 0px; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; -webkit-tap-highlight-color: transparent; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print {
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid; break-before: avoid; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  html.blink-to-pdf { font-size: 13px; }
  .typora-export #write { padding-left: 32px; padding-right: 32px; padding-bottom: 0px; break-after: avoid; }
  .typora-export #write::after { height: 0px; }
  @page { margin: 20mm 0px; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background: rgb(204, 204, 204); display: block; overflow-x: hidden; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
[contenteditable="true"]:active, [contenteditable="true"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: none; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) {
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: var(--monospace); }
code { text-align: left; vertical-align: initial; }
a.md-print-anchor { white-space: pre !important; border-width: initial !important; border-style: none !important; border-color: initial !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; background: 0px 0px !important; text-decoration: initial !important; text-shadow: initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.html-for-mac .inline-math-svg .MathJax_SVG { vertical-align: 0.2px; }
.md-math-block .MathJax_SVG_Display { text-align: center; margin: 0px; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; overflow-y: hidden; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: var(--monospace); }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: 400; line-height: normal; zoom: 90%; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; }
.MathJax_SVG * { transition: none; }
.MathJax_SVG_Display svg { vertical-align: middle !important; margin-bottom: 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="mermaid"] svg, [lang="flow"] svg { max-width: 100%; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom: 0px; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: none; }
.highlight td, .highlight tr { border: 0px; }


:root { --side-bar-bg-color: #fafafa; --control-text-color: #777; }
html { font-size: 16px; }
body { font-family: "Open Sans", "Clear Sans", "Helvetica Neue", Helvetica, Arial, sans-serif; color: rgb(51, 51, 51); line-height: 1.6; }
#write { max-width: 860px; margin: 0px auto; padding: 30px 30px 100px; }
#write > ul:first-child, #write > ol:first-child { margin-top: 30px; }
a { color: rgb(65, 131, 196); }
h1, h2, h3, h4, h5, h6 { position: relative; margin-top: 1rem; margin-bottom: 1rem; font-weight: bold; line-height: 1.4; cursor: text; }
h1:hover a.anchor, h2:hover a.anchor, h3:hover a.anchor, h4:hover a.anchor, h5:hover a.anchor, h6:hover a.anchor { text-decoration: none; }
h1 tt, h1 code { font-size: inherit; }
h2 tt, h2 code { font-size: inherit; }
h3 tt, h3 code { font-size: inherit; }
h4 tt, h4 code { font-size: inherit; }
h5 tt, h5 code { font-size: inherit; }
h6 tt, h6 code { font-size: inherit; }
h1 { padding-bottom: 0.3em; font-size: 2.25em; line-height: 1.2; border-bottom: 1px solid rgb(238, 238, 238); }
h2 { padding-bottom: 0.3em; font-size: 1.75em; line-height: 1.225; border-bottom: 1px solid rgb(238, 238, 238); }
h3 { font-size: 1.5em; line-height: 1.43; }
h4 { font-size: 1.25em; }
h5 { font-size: 1em; }
h6 { font-size: 1em; color: rgb(119, 119, 119); }
p, blockquote, ul, ol, dl, table { margin: 0.8em 0px; }
li > ol, li > ul { margin: 0px; }
hr { height: 2px; padding: 0px; margin: 16px 0px; background-color: rgb(231, 231, 231); border: 0px none; overflow: hidden; box-sizing: content-box; }
li p.first { display: inline-block; }
ul, ol { padding-left: 30px; }
ul:first-child, ol:first-child { margin-top: 0px; }
ul:last-child, ol:last-child { margin-bottom: 0px; }
blockquote { border-left: 4px solid rgb(223, 226, 229); padding: 0px 15px; color: rgb(119, 119, 119); }
blockquote blockquote { padding-right: 0px; }
table { padding: 0px; word-break: initial; }
table tr { border-top: 1px solid rgb(223, 226, 229); margin: 0px; padding: 0px; }
table tr:nth-child(2n), thead { background-color: rgb(248, 248, 248); }
table tr th { font-weight: bold; border-width: 1px 1px 0px; border-top-style: solid; border-right-style: solid; border-left-style: solid; border-top-color: rgb(223, 226, 229); border-right-color: rgb(223, 226, 229); border-left-color: rgb(223, 226, 229); border-image: initial; border-bottom-style: initial; border-bottom-color: initial; text-align: left; margin: 0px; padding: 6px 13px; }
table tr td { border: 1px solid rgb(223, 226, 229); text-align: left; margin: 0px; padding: 6px 13px; }
table tr th:first-child, table tr td:first-child { margin-top: 0px; }
table tr th:last-child, table tr td:last-child { margin-bottom: 0px; }
.CodeMirror-lines { padding-left: 4px; }
.code-tooltip { box-shadow: rgba(0, 28, 36, 0.3) 0px 1px 1px 0px; border-top: 1px solid rgb(238, 242, 242); }
.md-fences, code, tt { border: 1px solid rgb(231, 234, 237); background-color: rgb(248, 248, 248); border-radius: 3px; padding: 2px 4px 0px; font-size: 0.9em; }
code { background-color: rgb(243, 244, 244); padding: 0px 2px; }
.md-fences { margin-bottom: 15px; margin-top: 15px; padding-top: 8px; padding-bottom: 6px; }
.md-task-list-item > input { margin-left: -1.3em; }
@media print {
  html { font-size: 13px; }
  table, pre { break-inside: avoid; }
  pre { word-wrap: break-word; }
}
.md-fences { background-color: rgb(248, 248, 248); }
#write pre.md-meta-block { padding: 1rem; font-size: 85%; line-height: 1.45; background-color: rgb(247, 247, 247); border: 0px; border-radius: 3px; color: rgb(119, 119, 119); margin-top: 0px !important; }
.mathjax-block > .code-tooltip { bottom: 0.375rem; }
.md-mathjax-midline { background: rgb(250, 250, 250); }
#write > h3.md-focus::before { left: -1.5625rem; top: 0.375rem; }
#write > h4.md-focus::before { left: -1.5625rem; top: 0.285714rem; }
#write > h5.md-focus::before { left: -1.5625rem; top: 0.285714rem; }
#write > h6.md-focus::before { left: -1.5625rem; top: 0.285714rem; }
.md-image > .md-meta { border-radius: 3px; padding: 2px 0px 0px 4px; font-size: 0.9em; color: inherit; }
.md-tag { color: rgb(167, 167, 167); opacity: 1; }
.md-toc { margin-top: 20px; padding-bottom: 20px; }
.sidebar-tabs { border-bottom: none; }
#typora-quick-open { border: 1px solid rgb(221, 221, 221); background-color: rgb(248, 248, 248); }
#typora-quick-open-item { background-color: rgb(250, 250, 250); border-color: rgb(254, 254, 254) rgb(229, 229, 229) rgb(229, 229, 229) rgb(238, 238, 238); border-style: solid; border-width: 1px; }
.on-focus-mode blockquote { border-left-color: rgba(85, 85, 85, 0.12); }
header, .context-menu, .megamenu-content, footer { font-family: "Segoe UI", Arial, sans-serif; }
.file-node-content:hover .file-node-icon, .file-node-content:hover .file-node-open-state { visibility: visible; }
.mac-seamless-mode #typora-sidebar { background-color: var(--side-bar-bg-color); }
.md-lang { color: rgb(180, 101, 77); }
.html-for-mac .context-menu { --item-hover-bg-color: #E6F0FE; }
#md-notification .btn { border: 0px; }
.dropdown-menu .divider { border-color: rgb(229, 229, 229); }





 .typora-export li, .typora-export p, .typora-export,  .footnote-line {white-space: normal;} 
</style>
</head>
<body class='typora-export' >
<div  id='write'  class = 'is-node'><h1><a name='header-n0' class='md-header-anchor '></a>Check for fishy genes</h1><p>This human-intelligence training tasks - loop through each significant genes in the call-set and check for some unusual behavior. The major &quot;dubious&quot; aspects are -</p><ul><li><p>lie in a poorly-aligned region</p><ul><li>examples, with in-page references</li></ul></li><li><p>artifact in sequencing: the limited calls are from &quot;hypermutation patients&quot;</p><ul><li>examples, </li></ul></li><li><p>Not enough support</p></li></ul><p>All the potential fishy gene-cancer pairs are flagged for further checking with IGV. potential erroneous patient-wise mutation burden are also marked for further check.</p><p>marks in  the document</p><ul><li><strong>#fishy</strong> potentially problematic</li><li><strong>#interesting</strong> great pattern, uniqueness</li><li><strong>#broader</strong> broader cases, the support is not as strong</li><li><strong>#error</strong> gene symbol cannot be found</li><li><strong>#specific</strong> cancer-type specific</li><li><strong>#question</strong> simple thought on something ;) </li></ul><h2><a name='header-n31' class='md-header-anchor '></a>ACC: Adrenocortical carcinoma</h2><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/ACC_coMut.png' alt='comut-plot' referrerPolicy='no-referrer' /></p><ul><li>MEN1: overwhelming amount of nonsense mutations (and indels). (alternate chromosome structures and epigenetic gene regulation, encodes fro menin, a potential TSG)</li><li>C12orf65: This gene is very short (120bp), and the mutations seems to be scattered. <strong>#broader</strong> This gene encodes an mitochrondrial matrix protein that contributes to mitochondrial translation machinery, translation release factor activity.</li></ul><p>Three patients have simliar albeit different mutation spectrum from others.</p><h2><a name='header-n39' class='md-header-anchor '></a>BLCA: Bladder Urothelial Carcinoma</h2><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/BLCA_coMut.png' alt='plot' referrerPolicy='no-referrer' /></p><h4><a name='header-n41' class='md-header-anchor '></a>Patients</h4><p><strong>#fishy-patient</strong> check for the one with hypermutation.</p><h4><a name='header-n43' class='md-header-anchor '></a>Spectrum</h4><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/1561310115659.png' alt='1561310115659' referrerPolicy='no-referrer' /></p><p>The LEGO plot indicates a prevalence of C&gt;T and C&gt;G mutation, which agrees with our spectrum.</p><h4><a name='header-n46' class='md-header-anchor '></a>Genes</h4><ul><li><p>KDM6A: hallmark gene, modulates H3K27 acetylation and performs H3K27 demethylation, TSG+oncogene, leukemia-maintenance</p><ul><li>prevalence of truncation indels Q555* compared to missense! #surprise</li></ul></li><li><p>PIK3CA: hallmark, ras pathway</p></li><li><p>FBXW7: encodes a F-box protein that constitutes ubiquitin ligase complex, P-dependent ubiquitination</p></li><li><p>RHOB: <strong>#interesting</strong> there are a lot of missense in BLCA but not in any other cancers (E172K / P75T <proline>) <strong>#specific</strong> </p></li><li><p>ARID1A: long gene, hallmark (when depleted cannot go through cell-cycle arrest). </p></li><li><p>RB1: 900bp, negative regulator in cell cycle, mostly frameshift and non-sense.</p></li><li><p>CREBBP: 2250bp <strong>#broader</strong> transcriptional coactivation</p></li><li><p>KMT2C <strong>#error</strong> cannot find the symbol. The previous name MLL3 cannot be displayed (perhaps nothing?) and there are multiple forms of mutations spanning over many individuals <strong>#fishy</strong></p></li><li><p>KMT2B <strong>#error</strong> cannot be found - The previous name MLL2 cannot be displayed (perhaps nothing found?)</p></li><li><p>KMT2D <strong>#error</strong> same</p></li><li><p>STAG2: 1200, nonsense Q593*, cohesion complex at centromeres, mediates cell-type specific contacts between enhancers and promoters.</p></li><li><p>ELF3: evenly scattered mutation, short (350) <strong>#broader</strong></p></li><li><p>ERCC2: <strong>#specific</strong> short gene but very concentrated mutation spots - Y72C and N238S (<strong>#interesting</strong>, because N/S are chemically similar). In transcription-coupled nucleotide excision repair pathway.</p></li><li><p>CDKN2A: interact and stabilize P53, cell cycle G1 control</p><ul><li><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/G1-checkpoint.png' alt='Simplified diagram of how p53 halts the cell cycle at the G1/S checkpoint. p53 is activated by DNA damage and causes production of a Cdk inhibitor, which binds to the Cdk-G1/S cyclin complex and inactivates it. This halts the cell in G1 and prevents it from entering S phase, allowing time for the DNA damage to be fixed.' referrerPolicy='no-referrer' /></li></ul></li><li><p>GNA13 <strong>#specific</strong> R200G, Gprotein</p></li><li><p>CTNNB1<strong>#overlap</strong> gene product constitute adheren junctions,  which create and maintain epithelial cell layers by regulating cell growth and adhesion between cells.</p></li><li><p>ERBB3: mostly missense. Encodes a EGFR family tyrosine kinase, it can bind its ligand but not convey signal to cell through protein phosphorylation (no kinase domain).</p></li><li><p>ERBB2: S310Y S310F missense. This gene encodes a member of the epidermal growth factor (EGF) receptor family of receptor tyrosine kinases. This protein has no ligand binding domain of its own and therefore cannot bind growth factors. However, it does bind tightly to other ligand-bound EGF receptor family members to form a heterodimer, stabilizing ligand binding and enhancing kinase-mediated activation of downstream signalling pathways, such as those involving mitogen-activated protein kinase and phosphatidylinositol-3 kinase. (ERBB2 does not have ligand-binding domain while ERBB3 does not contain kinase domain)</p></li><li><p>OGDH: <strong>#fishy</strong> no clustering signal, 1kb, function mostly with energy metabolism, not a hallmark gene</p></li><li><p>PTPN4: protein tyrosine phosphatase, all missense, no clustering signal, potentially <strong>#fishy</strong> <online note:This RefSeq record was created from transcript and genomic sequence data to make the sequence consistent with the reference genome assembly. The genomic coordinates used for the transcript record were based on transcript alignments.></p></li><li><p>HIST1H3B: <strong>#error</strong> basic nuclear proteins that are responsible for nucleosome structure, no MC3 records found <strong>#fishy</strong></p></li><li><p>TMCO4: <strong>#overlap</strong> Q13* truncation. </p></li><li><p>TFPI2: clustered in R222C/R222H but not a hallmark gene. sparse elsewhere <strong>#interesting</strong></p></li><li><p>RBM26: no clustering , 1kb, <strong>#fishy</strong></p></li><li><p>RIPK4: <strong>#error</strong> nothing displayed in MC3, no signal from other callsets</p></li><li><p>ZBTB7B: codes for a zinc-finger containing transcription fatcor that acts as a key regulator of lineage commitment of immature T-cell precursors <strong>#broader</strong></p></li><li><p>PCDHGA8: <strong>#fishy</strong> sequencing artifacts (not saturated), no clustering</p></li><li><p>SF1: splicing factor 1: recognizes intron branch point sequence and is required for the early stages of spliceosome assembly</p></li><li><p>SF3B1: <strong>#spefic</strong> E902K in bladder (charge changes!) <strong>#interesting</strong> also splicing factors</p></li><li><p>PSIP1: <strong>#fishy</strong> sequencing artifacts, no clustering. hallmark: embryonic development, lysosomal statibility</p></li><li><p>ATAD5: scattered <strong>#fishy</strong> no hotspots, ATPase</p></li><li><p>AHR: aryl hydrocarbon receptor  - a ligand-activated tf to respond to aromatic hydrocarbons. Q383H presents in bladder / kidney / liver and lung. <strong>#interesting</strong></p></li><li><p>TRRAP: 3.6kb, potentially <strong>#fishy</strong> , many missense</p><ul><li><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/TRRAP.png' alt='1561319212821' referrerPolicy='no-referrer' /></li></ul></li><li><p>DIAPH2: no hotspots, perhaps over-representation, potentially <strong>#broader, #fishy</strong> 1kb</p><ul><li><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/DIAPH2.png' alt='1561319440421' referrerPolicy='no-referrer' /></li></ul></li><li><p>CUL1: E485K, ubiqitin protein ligase binding, activated TLR signaling, quite promising..</p></li><li><p>RXRA: <strong>#specific</strong> S427Y S427F retinoic acid-mediated gene activation. <del>retinoic acid related to bladder??</del></p></li><li><p>EP300: p300 transctiptional coactivator histone acetyltransferase via chromatin remodeling and important for proliferation and differentiation. 2.25kb Q182<strong>#specific</strong> truncation, bladder-specific mutation hotspots in 1100-1500bp</p></li></ul><h2><a name='header-n134' class='md-header-anchor '></a>BRCA: Breast cancer</h2><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/BRCA_coMut.png' alt='brca' referrerPolicy='no-referrer' /></p><p>‚ö†Ô∏è sequencing artifact in at least one individual!</p><p>üõë Individual ID: to be fillled</p><ul><li><p>PIK3CA: Classic! ras effector, catalytic subunit of kinase, many activating mutations üìéworth explore! Most missense mutation but there are inframe deletion and splice-site as well.</p><ul><li><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/rtk-pathway.jpg' alt='img' referrerPolicy='no-referrer' /></li></ul></li><li><p>TP53</p></li><li><p>AKT1: serine protein kinase. catalytically inactivate in serum-starved primary and immortalized fibroblasts. AKT1 and AKT2 are activated by platelet-derived growth factor. Intense hotspot in E17K.</p></li><li><p>CDH1(Cadherin1): <strong>#specific</strong> intense clustering of frameshift indel / nonsense in BRCA, but not other tumors! <strong>#interesting</strong> Q23*. cadherin, Ca-dependent cell cell adhesion glycoprotein. Increasing proliferatoin, invasion and metastasis.</p></li><li><p>CBFB: forms heterodimer with RUNX transcription factor to enhance their affinity for DNA, negatively regulated ribosomal gene expression during mitosis. Required for transcription of a subset of Runx2-target genes that are sufficient to maintain the invasive phenotype of the cells in breast cancer. (TSG/fusion) There are no specific mutation hotspot, but it seems to be an over-representation of CBFB mutation in BRCA compared to other cancers.</p></li><li><p>PTEN: seems the sequencing quality is not very good. classic TSG. <strong>#interesting Q:</strong> <del>why quality so low? special conformation??</del></p><ul><li><img src='https://www.researchgate.net/profile/Iain_Mcneish2/publication/8887434/figure/fig2/AS:277685025689615@1443216677879/The-Akt-PTEN-pathway-Oncogenic-and-mitogenic-stimuli-that-activate-PI3kinase-can-lead-to.png' alt='The Akt/PTEN pathway. Oncogenic and mitogenic stimuli that activate PI3kinase can lead to Akt activation, either directly, via the actions of phosphatidylinositol 3,4,5-trisphosphate (/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/pten-pathway.jpg ) and phosphati-√Ç ' referrerPolicy='no-referrer' /></li></ul></li><li><p>MAP3K1: mostly indel and truncating events in breast. The first 150 bp are error-free <del>could be vitally important??</del>! <strong>#broader</strong> but have literature support in breast.... hmmm. It is a mitogen activated protein kinase kinase kinase that activate JNK signalling pathway, stabilizes Myc by promoting N-terminal Phos and enhancing its transcriptional activity (Oncogene + TSG)</p><ul><li><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/MAP3K1.png' alt='1561324327829' referrerPolicy='no-referrer' /></li></ul></li><li><p>MAP2K4: potentially <strong>#fishy</strong> sequencing quality not good and it is not a hallmark gene. mutations are not clustered.</p><ul><li><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/MAP2K4.png' alt='1561324530318' referrerPolicy='no-referrer' /></li><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561325119350.png' alt='MAP2K4 with MAP3K1' referrerPolicy='no-referrer' /></li></ul></li><li><p>CASP8: encodes cysteine-aspartic acid protease (caspase) family, sequential activation of caspase plays central role in execution-phase of cell apoptosis. Q482H mutation of procaspase-8 abolishes caspase-8-mediated apoptosis by impairing procaspase-8 dimerization in AML.</p></li><li><p>FOXA1: oncogene hallmark. The major signal comes from prostate cancer cluster, it is very dense at 250bp! siliencing inhibits EMT and suppresses proliferation. (basically most findings come from gene perturbation? <del>we are bayesian</del>) </p></li><li><p>KMT2C <strong>#error</strong></p></li><li><p>KRAS: very clean plot! G12R/C, G13D, E62K, A146T spanning multiple cancers</p></li><li><p>SF3B1: mostly clusted in 600-720 region, but not a hallmark gene</p></li><li><p>ERBB2: as mentioned in BLCA</p></li><li><p>RNF5: potentially <strong>#fishy</strong> sequencing calls are not of good quality, not a hallmark gene</p><ul><li><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/portal_plots/RNF5.png' alt='1561326718061' referrerPolicy='no-referrer' /></li></ul></li><li><p>RB1: classic TSG, chromatin-associated protein that limits the transcription of cell cycle genes, primarily via regulation of the E2F transcription factor</p></li><li><p>ARID1A: TSG/fusion, classical, frequent inactivating mutations, fuse with MAST2 in triple-negative breast cancer. Maintains genome statibility, promote differentiation.</p></li><li><p>SMTNL2: potentially <strong>#fishy</strong> not enough support</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561327055800.png' alt='1561327055800' referrerPolicy='no-referrer' /></li></ul></li></ul><h2><a name='header-n195' class='md-header-anchor '></a>CESC: Cervical squamous cell carcinoma and endocervical adenocarcinoma</h2><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/CESC_coMut.png' alt='cesc' referrerPolicy='no-referrer' /></p><p>There are much more variation in mutation spectrum. ‚ö†Ô∏è check the first patient for sequencing issue!</p><ul><li><p>PIK3CA</p></li><li><p>FBXW7</p></li><li><p>EP300: coactivator lf HIF1A and thus plays a role in hypoxia-induced genes such as VEGF (<del>cervial cancer is related to hypoxia??</del>)</p></li><li><p>NFE2L2: nuclear factor like 2, encodes a TF which is a member of basic leucine zipper proteins, it regulates genes which contain antioxidant response elements (ARE) in their promoters. Very sharp peak! (it worths to check cervical cancer and oxygene related responses!)</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561327820111.png' alt='1561327820111' referrerPolicy='no-referrer' /></li></ul></li><li><p>KMT2C <strong>#error</strong></p></li><li><p>KRAS</p></li><li><p>KMT2D <strong>#error</strong></p></li><li><p>ERBB2</p></li><li><p>TP53</p></li><li><p>ARHGAP4: Rho GTPase activating protein 4, regulating proteins from Ras family. But the signals are weak - <strong>#broader</strong></p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561328018499.png' alt='1561328018499' referrerPolicy='no-referrer' /></li></ul></li></ul><h2><a name='header-n225' class='md-header-anchor '></a>CHOL: Cholangiocarcinoma</h2><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/CHOL_coMut.png' alt='chol' referrerPolicy='no-referrer' /></p><p>It seems the mutational burden for each CHOL patients are not as much, could be worth to check the patient with 12 muts(maybe not). The mutation spectrum are consistent.</p><ul><li>IDH1: isocitrate dehydrogenase 1 (NADP+), soluble, significant R132G/H, especially in brain lower grade glioma. Hallmarkgene: cytosolic krebs cycyle enzyme, catalyzing the formation of alpha-ketoglutarate from isocitrate. R132H alters histone marks and induces extensive DNA hypermethylation. Missense R132H causes a shift in the active site resulting in decreased binding affinity for NADPH. ‚Äã ‚ùì  <del>is CHOL pathologically related to energy metabolism?</del></li><li>BAP1: BRCA1 associated protein-1 (ubiquitin carboxy-terminal hydrolase) important in the G1/S transition, regulates the levels of cell cycle proteins by affecting E2F1-dependent S-phase gene expression</li><li>PBRM1: <strong>#interesting</strong> mostly nonsense. Similar scattered nonsense mutation observed in Kidney renal clear cell carcinoma. TSG chromatin remodeling complex. Mutation occurs mostly in bromodomain and BAH domain.</li></ul><h2><a name='header-n235' class='md-header-anchor '></a>COAD: Colon adenocarcinoma</h2><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/COAD_coMut.png' alt='COAD' referrerPolicy='no-referrer' /></p><p>‚ö†Ô∏è  There are lot of patients whose mutation burden is very high &gt;150 mutations/mb, they might be sequencing artifacts! The &quot;hyper-mutated&quot; patients have elevated number of C-&gt;A in T_T and A-&gt;G</p><ul><li><p>APC: intense signal from clustering of nonsense and inframe indels in colon adenocarcinoma (perhaps suggesting similar etiology). repressor of Wnt signaling pathway part of beta-catenin destruction complex, promotes cell migration, promotes apoptosis, classic hallmark gene, mutation mostly in unstructured protein sequence. </p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561329323273.png' alt='1561329323273' referrerPolicy='no-referrer' /></li><li><img src='https://www.researchgate.net/profile/Joyce_Ohm/publication/7286336/figure/fig4/AS:271656811757600@1441779439889/Addiction-to-the-Wnt-signalling-pathway-through-gene-silencing-eventsa-In-normal-colon.png' alt='Addiction to the Wnt signalling pathway through gene-silencing events.a | In normal colon epithelial cells, secreted frizzled-related proteins (SFRPs) function as antagonists of Wnt signalling by competing with Wnt proteins for binding to their receptor, Frizzled (FRZ). Expression of SFRPs is therefore the epigenetic gatekeeper step. When Wnt signalling is inactive, the adenomatosis polyposis coli (APC) complex phosphorylates -catenin, leading to its degradation. This prevents the accumulation of nuclear -catenin and therefore its ability to engage its transcription factor partners (TGF), which results in the differentiation and homeostasis of colon epithelial cells. Expression of APC is therefore a genetic gatekeeper step. b | When SFRP expression is lost, through epigenetic silencing of the gene that encodes it (loss of the epigenetic gatekeeper), Wnt signalling becomes activated through the receptor FRZ. This Wnt signalling potentially inactivates the APC complex (loss of the genetic gatekeeper), allowing -catenin to accumulate in the cytoplasm and eventually in the nucleus. In the nucleus, -catenin activates transcription of genes such as MYC, cyclin D and other genes whose products promote cell proliferation and survival rather than differentiation. This results in the expansion of colon epithelial stem and progenitor cells and formation of atypical crypt foci (ACF). c | Persistent activation of the Wnt pathway allows mutations to occur in other pathway components, such as those that permanently disable the APC complex and promote nuclear accumulation of -catenin (loss of the genetic gatekeeper, as indicated by the bold cross). These cells are selected for because of their survival and proliferative advantages. This combination of epigentic and genetic events fully activates the Wnt pathway to promote tumour progression. Without the epigenetic events that silence the SFRP genes, mutations that disrupt the APC complex might not be sufficient to promote tumorigenesis or tumour progression.' referrerPolicy='no-referrer' /></li></ul></li><li><p>TP53</p></li><li><p>KRAS</p></li><li><p>PIK3CA</p></li><li><p>BRAF</p></li><li><p>FBXW7</p></li><li><p>SMAD4</p></li><li><p>TCF7L2: transcription factor 7-like 2 (T-cell specific, HMG-box), Wnt signaling pathway, blood glucose homeostasis</p></li><li><p>PCBP1: <strong>#interesting #specific</strong> L100Q for colon cancer, KH domain, mrna splicing.</p></li><li><p>NRAS: signature G12D G13R Q61H/R/L/K</p></li><li><p>SMAD2:  This protein mediates the signal of the transforming growth factor (TGF)-beta, and thus regulates multiple cellular processes, such as cell proliferation, apoptosis, and differentiation.<strong>#interesting</strong> it has a consensus truncation at the end of transcript, (what is that for???)</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561346526779.png' alt='1561346526779' referrerPolicy='no-referrer' /></li></ul></li><li><p>AMER1: clustered truncation near 500 bp. R358*. binds to the tumour suppressor APC and acts as an inhibitor of Wnt signalling by inducing beta-catenin degradation <a href='https://www.ncbi.nlm.nih.gov/pubmed/21498506'>[Pubmed]<img src='https://cancer.sanger.ac.uk/core/gfx/blank.gif' alt='img' referrerPolicy='no-referrer' /></a>; inhibits the ubiquitination of NRF2; AMER1 and NRF2 compete for binding to KEAP1 <a href='https://www.ncbi.nlm.nih.gov/pubmed/22215675'>[Pubmed]<img src='https://cancer.sanger.ac.uk/core/gfx/blank.gif' alt='img' referrerPolicy='no-referrer' /></a>; enhances p53 acetylation by CBP/p300 [<a href='https://www.ncbi.nlm.nih.gov/pubmed/22285752'>Pubmed</a>]. hallmark TSG.</p></li><li><p>PDYN: prodynorphin, The protein encoded by this gene is a preproprotein that is proteolytically processed to form the secreted opioid peptides beta-neoendorphin, dynorphin, leu-enkephalin, rimorphin, and leumorphin. These peptides are ligands for the kappa-type of opioid receptor. Dynorphin is involved in modulating responses to several psychoactive substances, including cocaine. Mutations tends to happen at the end of protein. (<del>psychoactive substance with colon cancer???</del>) <strong>#broader</strong> </p></li><li><p>DAB2: <strong>#specific</strong> V172E disabled homolog 2, mitogen-responsive phosphoprotein, but it is not a classical hallmark</p></li><li><p>B2M: beta-2-microglobulin <strong>#question</strong> (why the frameshift indel is not flagged?) hallmark: recurrent missense mutations of the 1st amino acid, recurrent L15fs, inactivating mutations in 5&#39; region of the gene lead to impaired expression of HLA class I antigen, thus helping tumour to avoid the immune destruction in melanoma FO-1 cell line.</p></li><li><p>MAP2K4</p></li><li><p>CCDC30: <strong>#broader</strong> V396E, not classic hallmark</p></li><li><p>CHST15: glycosaminoglycan, This gene has also been identified as being co-expressed with RAG1 in B-cells and as potentially acting as a B-cell surface signaling receptor. <strong>#broader</strong> not well-supported, no obvious clustering, potentially <strong>#fishy</strong></p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561345520545.png' alt='1561345520545' referrerPolicy='no-referrer' /></li></ul></li><li><p>CCT6B: <strong>#fishy</strong> molecular chaperone to achieve optimal fold, not functionally-likely???? <strong>#broader</strong></p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561345735572.png' alt='1561345735572' referrerPolicy='no-referrer' /></li></ul></li><li><p>PTEN</p></li><li><p>ATM: classical hallmark. phosphorylates histone H2AX in response to DNA double-strand breaks <a href='https://www.ncbi.nlm.nih.gov/pubmed/11571274'>[Pubmed]<img src='https://cancer.sanger.ac.uk/core/gfx/blank.gif' alt='img' referrerPolicy='no-referrer' /></a>; plays a central role in the repair of DNA double-strand breaks, recruits and cooperates with other sensor proteins. Although  overall mutational burden is not very high (nor is it shows sign of clustering) we can believe the signals.</p></li><li><p>SNX13: <strong>#specific</strong> L663F Intracellular trafficking, G-protein signaling. <strong>#broader</strong></p></li><li><p>FAT4: FAT tumor suppressor homolog 4 (Drosophila) Hallmark TSG, protein belonging to cadherin (calcium-dependent cell adhesion protein) family <a href='https://www.ncbi.nlm.nih.gov/pubmed/23076869'>[Pubmed]<img src='https://cancer.sanger.ac.uk/core/gfx/blank.gif' alt='img' referrerPolicy='no-referrer' /></a>; suppresses phosphorylation and nuclear accumulation of Yap, a protein associated to the promoted proliferation, migration and cell cycle progression in gastric cancer.However the gene is very long 4.5kb and the mutation does not show any clustering. Therefore do not know much about colon-specific implications.</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561346244141.png' alt='1561346244141' referrerPolicy='no-referrer' /></li></ul></li><li><p>ASPM: Human ortholog of Drosophila abnormal spindle gene. mitotic spindle regulation, with a preferential role in regulating neurogenesis. Very long (3.5kb) high burden of syn mutations. <strong>#fishy</strong></p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561346426785.png' alt='1561346426785' referrerPolicy='no-referrer' /></li></ul></li><li><p>CYHR1: not enough information, no clustering mutation hotspots</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561346495312.png' alt='1561346495312' referrerPolicy='no-referrer' /></li></ul></li></ul><h2><a name='header-n312' class='md-header-anchor '></a>DLBC: Lymphoid Neoplasm Diffuse Large B-cell Lymphoma</h2><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/DLBC_coMut.png' alt='dlbc' referrerPolicy='no-referrer' /></p><p>‚ö†Ô∏è Per patient mutation burden relatively stable, quite some A-&gt;G and C-&gt;A mutations</p><p>The following genes have a expected tendency for immune function (features CDxxx), however will it be a result of higher power from overly expressed genes? <strong>#question</strong> The mutational plot for many genes here is sparse, but given the fact that they are mostly short protein - it could be expected.</p><ul><li><p>B2M</p></li><li><p>MYD88: This gene encodes a cytosolic adapter protein that plays a central role in the innate and adaptive immune response. This protein functions as an essential signal transducer in the interleukin-1 and Toll-like receptor signaling pathways. These pathways regulate that activation of numerous proinflammatory genes. <strong>#question</strong> why cosmic says the post prevalent substitution is L265* but we have most L252P (why is it labeled non/mis)??</p></li><li><p>TP53</p></li><li><p>BTG2: <strong>#specific but #fishy</strong>. However it is structually related to proteins that appear to have antiproliferative properties, involves in G1/S transition of cell cycle. <strong>#question</strong> why is this cluster only appear in DLBC, if the mechanism is so fundamental???</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561346996388.png' alt='1561346996388' referrerPolicy='no-referrer' /></li></ul></li><li><p>TMSB4X: really short!!! an actin sequestering protein which plays a role in regulation of actin polymerization. The protein is also involved in cell proliferation, migration, and differentiation. This gene escapes X inactivation and has a homolog on chromosome Y. But viewing from the plot, we do not have enough confidence due to lack of support.<strong>#fishy</strong></p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561347197852.png' alt='1561347197852' referrerPolicy='no-referrer' /></li></ul></li><li><p>IGLL5: many syn mutation, potentially <strong>#fishy</strong> Not much support from literature</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561347297861.png' alt='1561347297861' referrerPolicy='no-referrer' /></li></ul></li><li><p>KLHL6: This gene encodes a member of the kelch-like (KLHL) family of proteins, which is involved in B-lymphocyte antigen receptor signaling and germinal-center B-cell maturation. The encoded protein contains an N-terminal broad-complex, tramtrack and bric a brac (BTB) domain that facilitates protein binding and dimerization, a BTB and C-terminal kelch (BACK) domain, and six C-terminal kelch repeat domains. Naturally occurring mutations in this gene are associated with chronic lymphocytic leukemia. <strong>#broader</strong> there is no clustering and the background syn rate is high</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561347744236.png' alt='1561347744236' referrerPolicy='no-referrer' /></li></ul></li><li><p>UBE2A: ubiquitin pathway - The modification of proteins with ubiquitin is an important cellular mechanism for targeting abnormal or short-lived proteins for degradation. Ubiquitination involves at least three classes of enzymes: ubiquitin-activating enzymes, or E1s, ubiquitin-conjugating enzymes, or E2s, and ubiquitin-protein ligases, or E3s. This gene encodes a member of the E2 ubiquitin-conjugating enzyme family. This enzyme is required for post-replicative DNA damage repair. However the mutation is very sparse and lacking clustering - it worth to note that we do not have syn here - only splice and nonsense. <strong>#broader</strong></p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561347902537.png' alt='1561347902537' referrerPolicy='no-referrer' /></li></ul></li><li><p>CD70: The protein encoded by this gene is a cytokine that belongs to the tumor necrosis factor (TNF) ligand family. This cytokine is a ligand for TNFRSF27/CD27. It is a surface antigen on activated, but not on resting, T and B lymphocytes. It induces proliferation of costimulated T cells, enhances the generation of cytolytic T cells, and contributes to T cell activation. This cytokine is also reported to play a role in regulating B-cell activation, cytotoxic function of natural killer cells, and immunoglobulin sythesis. <strong>#broader</strong> mutations are so sparse</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561348160032.png' alt='1561348160032' referrerPolicy='no-referrer' /></li></ul></li><li><p>CD79B: <strong>#specific</strong> A68N for DLBC (A196N) <strong>#question</strong> different gene model? Ig beta, activator of Syk kinase <a href='https://www.ncbi.nlm.nih.gov/pubmed/15219998'>[Pubmed]<img src='https://cancer.sanger.ac.uk/core/gfx/blank.gif' alt='img' referrerPolicy='no-referrer' /></a>; truncating mutation in Ig beta prevents the assembly of the IgM BCR on the cell surface, resulting in the absence of peripheral B cells and low/absent immunoglobulin serum levels <a href='https://www.ncbi.nlm.nih.gov/pubmed/17709424'>[Pubmed]<img src='https://cancer.sanger.ac.uk/core/gfx/blank.gif' alt='img' referrerPolicy='no-referrer' /></a></p></li><li><p>PIM1: The protein encoded by this gene belongs to the Ser/Thr protein kinase family, and PIM subfamily. This gene is expressed primarily in B-lymphoid and myeloid cell lines, and is overexpressed in hematopoietic malignancies and in prostate cancer. It plays a role in signal transduction in blood cells, contributing to both cell proliferation and survival, and thus provides a selective advantage in tumorigenesis. Both the human and orthologous mouse genes have been reported to encode two isoforms (with preferential cellular localization) resulting from the use of alternative in-frame translation initiation codons, the upstream non-AUG (CUG) and downstream AUG codons. </p></li><li><p>FAS: TNF receptor superfamily. The interaction of this receptor with its ligand allows the formation of a death-inducing signaling complex that includes Fas-associated death domain protein (FADD), caspase 8, and caspase 10. The autoproteolytic processing of the caspases in the complex triggers a downstream caspase cascade, and leads to apoptosis. This receptor has been also shown to activate NF-kappaB, MAPK3/ERK1, and MAPK8/JNK, and is found to be involved in transducing the proliferating signals in normal diploid fibroblast and T cells. Several alternatively spliced transcript variants have been described, some of which are candidates for nonsense-mediated mRNA decay (NMD). The isoforms lacking the transmembrane domain may negatively regulate the apoptosis mediated by the full length isoform.</p></li><li><p>HLA-B <strong>#error</strong> nothing displayed</p></li><li><p>CARD11: molecular scaffolds, This protein is also a member of the CARD protein family, which is defined by carrying a characteristic caspase-associated recruitment domain (CARD). This protein has a domain structure similar to that of CARD14 protein. The CARD domains of both proteins have been shown to specifically interact with BCL10, a protein known to function as a positive regulator of cell apoptosis and NF-kappaB activation. When expressed in cells, this protein activated NF-kappaB and induced the phosphorylation of BCL10. <strong>#broader</strong></p></li><li><p>KMT2D <strong>#error</strong></p></li></ul><h2><a name='header-n365' class='md-header-anchor '></a>ESCA: Esophageal carcinoma</h2><p>&nbsp;</p><p><img src='/home/qingzhang/Downloads/comut_pack/comut_plots_pngs/ESCA_coMut.png' alt='esca' referrerPolicy='no-referrer' /></p><p>Two patients are significantly hypermutated. </p><ul><li><p>TP53</p></li><li><p>NFE2L2: Mostly missense. This gene encodes a transcription factor which regulates genes which contain antioxidant response elements (ARE) in their promoters; many of these genes encode proteins involved in response to injury and inflammation which includes the production of free radicals. (<strong>#question</strong> is there some mutational signatures associated with free radicals?)</p></li><li><p>CDKN2A: inhibitor of CDK4 kinase. cell cycle G1 control. </p></li><li><p>SMAD4</p></li><li><p>PIK3CA</p></li><li><p>AP2B1: protein product links clathrin to receptors in coated vesicles potentially <strong>#fishy</strong>  - could be sequencing artifacts</p><ul><li><img src='/home/qingzhang/.config/Typora/typora-user-images/1561349321459.png' alt='1561349321459' referrerPolicy='no-referrer' /></li></ul></li></ul><h2><a name='header-n385' class='md-header-anchor '></a>GBM</h2><ul><li>TP53</li><li>PTEN</li><li>EGFR</li><li>PIK3CA</li><li>RB1</li><li>NF1</li><li>IDH1</li><li>PIK3R1</li><li>BRAF</li><li>PDGFRA</li><li>MAP3K1</li><li>SEMA3C</li><li>SEMG1</li><li>SLC26A3</li><li>EXOC2</li><li>LZTR1</li><li>COL1A2</li><li>KRT15</li></ul><h2><a name='header-n423' class='md-header-anchor '></a>HNSC</h2><ul><li>TP53</li><li>PIK3CA</li><li>CDKN2A</li><li>FAT1</li><li>NOTCH1</li><li>CASP8</li><li>NSD1</li><li>EP300</li><li>HRAS</li><li>TGFBR2</li><li>KMT2D</li><li>HLA-B</li><li>RAC1</li><li>FBXW7</li><li>RHOA</li><li>ZNF750</li><li>RB1</li><li>KEAP1</li><li>EPHA2</li><li>FGFR3</li><li>MAPK1</li><li>MB21D2</li><li>NFE2L2</li><li>ZNF623</li><li>DOK6</li><li>PTEN</li><li>EEF1A1</li><li>NCOR1</li><li>LCP1</li><li>CCDC50</li><li>ADCY1</li><li>GIGYF2</li><li>CUL3</li></ul><h2><a name='header-n491' class='md-header-anchor '></a>KICH</h2><p>TP53</p><h2><a name='header-n493' class='md-header-anchor '></a>KIRC</h2><ul><li>MTOR</li><li>BAP1</li><li>VHL</li><li>PBRM1</li><li>STED2</li><li>SETD2</li><li>TP53 (why so less important)</li><li>PIK3CA</li><li>PTEN</li><li>ATM</li></ul><h2><a name='header-n515' class='md-header-anchor '></a>KIRP</h2><ul><li>MET</li><li>KRAS</li></ul><h2><a name='header-n521' class='md-header-anchor '></a>LGG</h2><ul><li>IDH1</li><li>TP53</li><li>ATRX</li><li>CIC</li><li>PIK3CA</li><li>EGFR</li><li>IDH2</li><li>PTEN</li><li>FUBP1</li><li>NOTCH1</li><li>ZBTB20</li><li>PIK3R1</li><li>SMARCA4</li><li>NF1</li><li>NRAS</li><li>PTPN11</li><li>DNMT3A</li></ul><h1><a name='header-n557' class='md-header-anchor '></a>General Questions</h1><h2><a name='header-n558' class='md-header-anchor '></a>chromosome translocations</h2><p>mutation happens on an unstructed protein segments or a well-characterized one? bent is more vulnerable?</p><p>if truncating events and frameshift prevails, what is it implies? very obvious selection advantage?</p><h1><a name='header-n561' class='md-header-anchor '></a>General pathways</h1><h3><a name='header-n562' class='md-header-anchor '></a>RAS signaling</h3><p>&nbsp;</p></div>
</body>
</html>